alkira/resource_alkira_flow_collector.go

```diff
@@ -0,0 +1,250 @@
+package alkira
+
+import (
+	"context"
+	"fmt"
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAlkiraFlowCollector() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provide flow collector resource.",
+		CreateContext: resourceFlowCollector,
+		ReadContext:   resourceFlowCollectorRead,
+		UpdateContext: resourceFlowCollectorUpdate,
+		DeleteContext: resourceFlowCollectorDelete,
+		CustomizeDiff: func(ctx context.Context, d *schema.ResourceDiff, m interface{}) error {
+			client := m.(*alkira.AlkiraClient)
+
+			old, _ := d.GetChange("provision_state")
+
+			if client.Provision == true && old == "FAILED" {
+				d.SetNew("provision_state", "SUCCESS")
+			}
+
+			return nil
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the flow collector.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "The description of the flow collector.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"provision_state": {
+				Description: "The provisioning state of the resource.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"collector_type": {
+				Description:  "The type of the flow collector.",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "GENERIC",
+				ValidateFunc: validation.StringInSlice([]string{"GENERIC"}, false),
+			},
+			"enabled": {
```

Comment:
I think we usually have a default value for this?

Reply:
this one is user choice as this can cause them to start billing on the flow collector. User can create and keep it disabled and turn it on only when they want to collect flows.